### PR TITLE
Unit tests support for sis

### PR DESCRIPTION
**Pull request by Xarpagon**

### CODE REVIEW

File: src/leon3.c [hunk 1]
```diff
@@ -53,10 +53,18 @@
 /* APB UART registers */
 
 #define APBUART0_ADDRESS    0x80000100
-#define APBUART0_IRQ 2
+// #define APBUART1_ADDRESS    0x80100100
```
Comment (Lurkerpas) [inline src/leon3.c:56]: Why are these commented out?

File: src/leon3.c [hunk 2]
```diff
@@ -86,7 +94,11 @@ init_sim (void)
   grlib_ahbs_add (&sdctrl, 0, RAM_START, RAM_MASKPP);
 
   grlib_apb_add (&apbuart, APBUART0_IRQ, APBUART0_ADDRESS, 0xFFF);
-  grlib_apb_add (&apbuart, APBUART1_IRQ, APBUART1_ADDRESS, 0xFFF);
+  // grlib_apb_add (&apbuart, APBUART1_IRQ, APBUART1_ADDRESS, 0xFFF);
```
Comment (Lurkerpas) [inline src/leon3.c:97]: Why are those commented out?

File: test/unit/main.c [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#include "uart/uart_read_uart.h"
```
Comment (Lurkerpas) [inline test/unit/main.c:1]: How does this relate to the MR title?

File: test/unit/main.c [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
```
Comment (Lurkerpas) [inline test/unit/main.c:2]: Why not use a unit testing framework?

**Review by Lurkerpas — CHANGES_REQUESTED** [review]

File: test/unit/main.c (hunk 3 above)
Comment (Xarpagon) [reply test/unit/main.c]: Resolved.

File: test/unit/main.c (hunk 4 above)
Comment (Xarpagon) [reply test/unit/main.c]: CppUTest added.

File: .github/workflows/workflow.yml [hunk 5]
```diff
@@ -17,5 +17,8 @@ jobs:
     - name: Build
       run: make all
 
+    - name: Unit testing
+      run: ./build/test/unit/test -c
```
Comment (Lurkerpas) [inline .github/workflows/workflow.yml:21]: Why isn't this run via make check? If we want to have separate unit and integration tests, they can have separate targets (make unittest; make integrationtest), which are combined via make check.

Comment (Xarpagon) [reply .github/workflows/workflow.yml]: ack

**Review by Lurkerpas — APPROVED** [review]

**Pull request merged**